### ISSUE TITLE
[docs] Add `aria-label`, `aria-labelledby` and `aria-describedby` guidance to Data Grid Accessibility page

### ISSUE THREAD

**Issue by KenanYusuf** [mui/mui-x#17025]
### Search keywords

n/a

### Related page

https://mui.com/x/react-data-grid/accessibility/

### Kind of issue

Missing information

### Issue description

We should include guidance on the following—taken from the [Grid APG pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/)—as part of our Data Grid - Accessibility page:

> If there is an element in the user interface that serves as a label for the grid, [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) is set on the grid element with a value that refers to the labelling element. Otherwise, a label is specified for the grid element using [aria-label](https://w3c.github.io/aria/#aria-label).
> If the grid has a caption or description, [aria-describedby](https://w3c.github.io/aria/#aria-describedby) is set on the grid element with a value referring to the element containing the description.

### Context

Related to #17024
